obspy/io/hypodd/pha.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+"""
+HypoDD PHA read support.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+import io
+
+from obspy import UTCDateTime
+from obspy.core.event import (
+    Catalog, Event, Origin, Magnitude, Pick, WaveformStreamID, Arrival,
+    OriginQuality)
+
+
+def _seed_id_map(inventory=None, id_map=None, id_default='.{}..{}'):
+    if id_map is None:
+        id_map = {}
+    ret = id_map.copy()
+    if inventory is not None:
+        for net in inventory:
+            for sta in net:
+                if len(sta) == 0:
+                    temp = id_map.get(sta.code, id_default)
```

Comment:
I'm a bit confused here - if the station is already in the id_map the users have given some mapping explicitly (they might want to for example map the station to a different network code). I'd rather just `continue` here as user specified things should not be overwritten. But I might also just be missing something here.

Reply:
I think you are right. I will change this. I wanted to also support inventories with only network and station information, but that does not make much sense.

I think I will also add a warning when the construction of the id_map would not be unique (e.g. BH and HH channels in inventory) and move these functions to some util module, because they are also used by evt plugin. Is it OK if I add a new file obspy/io/util.py?